core/aws-core/src/main/java/software/amazon/awssdk/awscore/client/builder/AwsDefaultClientBuilder.java

```diff
@@ -433,4 +442,22 @@ public final BuilderT defaultsMode(DefaultsMode defaultsMode) {
     public final void setDefaultsMode(DefaultsMode defaultsMode) {
         defaultsMode(defaultsMode);
     }
+
+    private boolean idHasFipsPattern(Region region) {
+        return region.id().startsWith(FIPS_PREFIX) || region.id().endsWith(FIPS_SUFFIX);
+    }
+
+    private Region removeFips(Region region) {
+        if (region.id().startsWith(FIPS_PREFIX)) {
+            String prefixRemoved = StringUtils.replaceOnce(region.id(), FIPS_PREFIX, "");
+            return Region.of(prefixRemoved);
+        }
+
+        if (region.id().endsWith(FIPS_SUFFIX)) {
+            String suffixRemoved = StringUtils.replaceOnce(region.id(), FIPS_SUFFIX, "");
+            return Region.of(suffixRemoved);
+        }
```

Comment:
are we supposed to clean double-fips stuff? fips-test-1-fips -> test-1?

Reply:
The spec doesn't call this out specifically, nor do the test cases. I'll assume no for now. It *does* call out some infix cases which I'll fix.